lib/exrethinkdb/connection.ex

```diff
@@ -8,29 +19,38 @@ defmodule Exrethinkdb.Connection do
   end
 
   def init(opts) do
-    host = Dict.get(opts, :host, 'localhost')
+    host = Dict.get(opts, :host, {127,0,0,1})
     port = Dict.get(opts, :port, 28015)
-    socket = connect(host, port)
+    auth_key = Dict.get(opts, :auth_key, "")
+    socket = connect(host, port,auth_key)
     :ok = :inet.setopts(socket, [active: true])
     {:ok, %{pending: %{}, current: {:start, ""}, socket: socket, token: 0}}
   end
 
-  defp connect(host, port) do
+  defp connect(host, port,auth_key) do
     host = case host do
        x when is_binary(x) -> String.to_char_list x
        x -> x
     end
     {:ok, socket} = :gen_tcp.connect(host, port, [active: false, mode: :binary])
-    :ok = handshake(socket)
+    :ok = handshake(socket,auth_key)
     socket
   end
 
-  defp handshake(socket) do
-    :ok = :gen_tcp.send(socket, << 0x400c2d20 :: little-size(32) >>)
-    :ok = :gen_tcp.send(socket, << 0 :: little-size(32) >>)
-    :ok = :gen_tcp.send(socket, << 0x7e6970c7 :: little-size(32) >>)
-    {:ok, "SUCCESS" <> << 0 :: size(8)  >>} = :gen_tcp.recv(socket, 8)
-    :ok
+  defp handshake(socket,auth_key) do
+        :ok = :gen_tcp.send(socket, :binary.encode_unsigned(@proto_version,:little))
```

Comment:
encode_unsigned doesn't guarantee 4 bytes, which is what the spec wants. Unless I'm misunderstanding it.